stub-connector/src/main/java/uk/gov/ida/notification/stubconnector/resources/MetadataResource.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.ida.notification.stubconnector.resources;
+
+import org.opensaml.core.xml.XMLObject;
+import org.opensaml.core.xml.util.XMLObjectSupport;
+import uk.gov.ida.notification.stubconnector.metadata.MetadataGenerator;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+
+@Path("/ConnectorMetadata")
+@Produces(MediaType.APPLICATION_XML)
+public class MetadataResource {
+
+    private final MetadataGenerator metadataGenerator;
+
+    public MetadataResource(MetadataGenerator metadataGenerator) {
+        this.metadataGenerator = metadataGenerator;
+    }
+
+    @GET
+    public Response getConnectorNodeMetadata() throws Exception {
+        XMLObject xmlObject = metadataGenerator.getConnectorMetadata();
```

Comment:
Rather than regenerate and sign the metadata every time this endpoint is hit, could it be cached? Or just generate it once when the `MetadataResource` is initialised, with a really long validity period?

Reply:
As this is the stub connector I wasn't too concerned with caching, but I'm going to add a `@CacheControl` annotation to simulate a cache. Also adding a metadata expiry env var so we can simulate a stub country with expired metadata.

Reply:
I guess I was thinking more that the metadata isn't going to change or be regenerated once the app is up and running, so to save processing does it make sense to just generate the metadata once when the app boots up, rather than doing it every time the endpoint is hit. The cache you've added will help but it might be better to just only do it once but with a long validity.

Having said this, the amount it'll get used means this probably doesn't matter at all.